aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -189,12 +192,53 @@ struct AbsMinOps {
 #endif
 };
 
+template <typename scalar_t>
+inline C10_DEVICE scalar_t min_propagate_nan(scalar_t a, scalar_t b) {
+  if (at::_isnan(a) || at::_isnan(b)) {
+    return std::numeric_limits<scalar_t>::quiet_NaN();
+  } else {
+    return MIN(a, b);
+  }
+}
+
+// This accumulator template is used to calculate the minimum absolute value of
+// a set of numbers, with NaN value propagation.
+// `scalar_t` is the type of the input and `acc_t` is the type of the accumulated
+// value. These types differ for complex number input support.
+template <typename scalar_t, typename acc_t=scalar_t>
+struct AbsMinOps {
+
+  inline C10_DEVICE acc_t reduce(acc_t acc, scalar_t data, int64_t /*idx*/) const {
+    return min_propagate_nan<acc_t>(acc, static_cast<acc_t>(std::abs(data)));
+  }
+
+  inline C10_DEVICE acc_t combine(acc_t a, acc_t b) const {
+    return min_propagate_nan<acc_t>(a, b);
+  }
+
+  inline C10_DEVICE acc_t project(acc_t a) const {
+    return a;
+  }
+
+  static C10_DEVICE acc_t translate_idx(acc_t acc, int64_t /*base_idx*/) {
+    return acc;
+  }
+
+#if defined(__CUDACC__) || defined(__HIPCC__)
+  inline C10_DEVICE acc_t warp_shfl_down(acc_t acc, int offset) const {
+    return WARP_SHFL_DOWN(acc, offset);
+  }
+#endif
+};
+
 // This accumulator template is used to calculate the maximum absolute value of
 // a set of numbers.
 // `scalar_t` is the type of the input and `acc_t` is the type of the accumulated
 // value. These types differ for complex number input support.
+// TODO: Once `torch.norm` is removed and replaced with `torch.linalg.norm`,
```

Comment:
Add the reason this is "Old." Would "AbsMaxOpsDeprecated" be more accurate? 

Instead of deprecating this, however, is there a world where we want AbsMaxOpsIgnoreNaN? NumPy has a lot of "nan-" reductions, and SciPy has arguments for controlling how NaNs are treated.

Reply:
I like the AbsMaxOpsIgnoreNaN suggestion, there could be a plausible reason to keep using this, even if `torch.norm` is removed

Reply:
Incorrect nan behavior currently is due to bug in MIN/MAX implementation https://github.com/pytorch/pytorch/blob/b2d8f0a431f8ce2c606ed25ac4d28287489c71bc/aten/src/ATen/native/SharedReduceOps.h#L23-L29 on cuda only. We should fix it and have single version of AbsMaxOps.

Reply:
I suppose we can always add back AbsMaxOpsIgnoreNaN later if we need it ;)

I spoke to @ngimel offline, @kurtamohler, and if it's OK with you we would like to fix the NaN propagation behavior in this PR. We'll need to add a test for that change for torch.norm, too. For the complex backward issue, that affects torch.norm, too, right? Would you file an issue for that incorrect torch.norm behavior? Whether we address that issue may depend on whether we actually deprecate torch.norm before the 1.9 release or not.

Reply:
Got it, sounds good. Yeah I believe torch.norm's complex backward would be affected, but I'll double check first